### ISSUE TITLE
Truncate all minified files by default

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-code-manager#1462]
~~Depends on #1495~~

This was discussed in a meeting on Feb. 26, 2020. There is no need for a reviewer to see the full contents of a minified file, and it is known that minified files cause performance issues for the code manager, both in Browse and in Compare mode. While this will not solve the performance issue for all large files, it is seen as a quick win which is fairly easy to implement and will produce an immediate improvement.

We will use the info generated by the linter to identify a file as minified. It is possible that this could be a false positive, but in these cases a reviewer can simply click the link that allows them to view the whole file.

This will require a change to the API to inform the front end that a file is minified.

For QA:

After this lands, all minified files should be trimmed by default in the Browse and Compare views, regardless of their size. Note that, prior to this patch, in the Compare view, files *were not* trimmed by default, even if they are considered too large. After this patch lands all minified files *will be* trimmed by default in the Compare view.

**Comment by ioanarusiczki** [comment]
@bobsilverberg 
I've been checking on dev with FF74(Win10) so test results are here:
- minified files over 2000 chars show the browse page and compare page being trimmed by  default. The browse page is trimmed with syntax highlighting and the compare page is trimmed and the syntax highlighting is disabled. 
Examples:
https://code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?allowSlowPages=false&path=src%2Fcontent%2Fengines%2FalibabaChina.js
https://code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?allowSlowPages=false&path=src%2Fcontent%2Fengines%2FalibabaChina.js

- I have an example of a minified file of 390 chars which has the same length with both allowSlowPages=true or allowSlowPages=false. So I don't know if that's alright or it has to do with what you've said above 

> After this lands, all minified files should be trimmed by default in the Browse and Compare views, regardless of their size.

https://code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?allowSlowPages=false&path=src%2Fcontent%2Fengines%2FalibabaChina.js

**Comment by bobsilverberg** [comment]
@ioanarusiczki I see what's going on. When we trim a file, we reduce it down to 2000 characters, so if the file is already less than 2000 characters then trimming it actually does nothing. The message is still being displayed, however, that says the file was trimmed, because we *tried* to trim it, but trimming it yielded nothing. Can you please open a separate issue for minified files that are less than 2000 characters? We should not be displaying the message if the file was not in fact trimmed.

**Comment by ioanarusiczki** [comment]
@bobsilverberg I think I found a file which looks minified to me but the response in the network shows it's not uses_unknown_minified_code : false and that's perhaps the reason why it's not trimmed in compare view. Should I file an issue on server?

**Comment by bobsilverberg** [comment]
> @bobsilverberg I think I found a file which looks minified to me but the response in the network shows it's not uses_unknown_minified_code : false and that's perhaps the reason why it's not trimmed in compare view. Should I file an issue on server?
> 
> Two examples
> 
> [code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?path=src%2Faction%2Fstyle.css.map](https://code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?path=src%2Faction%2Fstyle.css.map)
> 
> [code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?path=src%2Faction%2Fstyle.css](https://code.addons-dev.allizom.org/en-US/compare/618937/versions/1690424...1690425/?path=src%2Faction%2Fstyle.css)

I'm not sure how accurate the algorithm is that identifies minified files, but yes, filing an issue for addons-server would be good in case it can be improved to identify more types of files.

**Comment by bobsilverberg** [comment]
> I'm not sure how accurate the algorithm is that identifies minified files, but yes, filing an issue for addons-server would be good in case it can be improved to identify more types of files.

@ioanarusiczki were you going to file an issue for this? I don't see one.

**Comment by ioanarusiczki** [comment]
@bobsilverberg I did not have time to further investigate other file types. But I can file the issue with the examples above and I'll try to find more examples to add.

**Comment by ioanarusiczki** [comment]
@bobsilverberg Done https://github.com/mozilla/addons-server/issues/13877
